bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/ScenarioHandler.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.boschshc.internal.devices.bridge;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.HttpStatus;
+import org.openhab.binding.boschshc.internal.devices.bridge.dto.Scenario;
+import org.openhab.binding.boschshc.internal.serialization.GsonUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * Handler for executing a scenario.
+ *
+ * @author Patrick Gell - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class ScenarioHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private final Map<String, Scenario> availableScenarios;
+
+    protected ScenarioHandler(Map<String, Scenario> availableScenarios) {
+        this.availableScenarios = Objects.requireNonNullElseGet(availableScenarios, HashMap::new);
+    }
+
+    public void executeScenario(final @Nullable BoschHttpClient httpClient, final String scenarioName) {
+        assert httpClient != null;
+        if (!availableScenarios.containsKey(scenarioName)) {
```

Comment:
I'm not sure whether caching scenarios by name is a good strategy. It would work as long the user does not delete a scenario and then creates a scenario with the same name (and a different ID) again. Of course it is an edge case but maybe we can find another solution (either a different caching strategy or obtaining scenarios eagerly every time).

Have you experimented with triggering scenarios without caching them? Is there a noticeable delay or performance issue if we don't use a cache?

Reply:
I did made a simple performance comparison and the difference was just a few milliseconds as you can see in the following diagramm (blue = withOUT cache, red = with cache)
<img width="992" alt="Screenshot 2023-11-04 at 07 27 37" src="https://github.com/openhab/openhab-addons/assets/54861416/e9616856-c60c-43c4-a5c1-a41cf05d4688">
Therefore I removed the caching